client/src/components/Account/EditProfileForm/EditProfileForm.tsx

```diff
@@ -1,52 +1,156 @@
-import { Box, Button, Grid, Typography } from '@material-ui/core';
+import { Box, Button, Grid, Typography, TextField } from '@material-ui/core';
+import Grow from '@material-ui/core/Grow';
 
 import useStyles from './useStyles';
 import CustomTextField from './CustomTextField';
 
+import React, { useState, useEffect } from 'react';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+
+import updateProfile from '../../../helpers/APICalls/updateProfile';
+import { OwnerFormProfile } from '../../../interface/Profile';
+
 export default function EditProfileForm(): JSX.Element {
   const classes = useStyles();
-  const handleChange = () => null;
+  const { loggedInUserDetails, updateLoggedInUserDetails } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const [profile, setProfile] = useState<OwnerFormProfile>({
+    firstName: '',
+    lastName: '',
+    email: '',
+    phoneNumber: '',
+    address: '',
+    description: '',
+  });
+  useEffect(() => {
+    if (loggedInUserDetails !== undefined) {
+      setProfile({
+        ...profile,
+        firstName: loggedInUserDetails?.firstName,
+        lastName: loggedInUserDetails?.lastName,
+        email: loggedInUserDetails?.email,
+        phoneNumber: loggedInUserDetails?.phoneNumber,
+        address: loggedInUserDetails?.address,
+        description: loggedInUserDetails?.description,
+      });
+    }
+  }, [loggedInUserDetails]);
+
+  const [showPhoneInput, setShowPhoneInput] = useState(!profile?.phoneNumber ? true : false);
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>,
+    property: string,
+  ): void => {
+    setProfile({ ...profile, [property]: e.target.value });
+  };
+
+  const toggleInput = () => setShowPhoneInput(!showPhoneInput);
+
+  const handleSaveProfile = async () => {
+    const id = loggedInUserDetails ? loggedInUserDetails._id : '';
+    try {
+      const res = await updateProfile(id, profile);
+      updateLoggedInUserDetails(res);
+      updateSnackBarMessage('Profie updated');
+    } catch (error) {
+      updateSnackBarMessage(`Error updating user profile ${error}`);
+    }
+  };
+
   return (
     <Box>
       <Typography variant="h4" align="center" className={classes.formTitle}>
         Edit Profile
       </Typography>
       <form>
         <Grid container spacing={3}>
-          <CustomTextField onChange={handleChange} value="" label="FIRST NAME" placeholder="First Name" />
-          <CustomTextField onChange={handleChange} value="" label="LAST NAME" placeholder="Last Name" />
-          <CustomTextField onChange={handleChange} value="" label="EMAIL ADDRESS" placeholder="user@gmail.com" />
+          <CustomTextField
+            onChange={(e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>): void =>
+              handleChange(e, 'firstName')
```

Comment:
as a suggestion, you can also introduce a property maybe called "name" to `CustomTextField` and give it the value of `firstName` (or whatever appropriate), that way you can probably reduce on repeated code throughout this file and it'll just adapt to handleChange, reading the "name" of that :)